driver-core/src/main/java/com/datastax/driver/core/Responses.java

```diff
@@ -388,7 +387,9 @@ public static Metadata decode(ByteBuf body, boolean withPkIndices) {
                         String ksName = globalTablesSpec ? globalKsName : CBUtil.readString(body);
                         String cfName = globalTablesSpec ? globalCfName : CBUtil.readString(body);
                         String name = CBUtil.readString(body);
-                        DataType type = DataType.decode(body);
+                        // will set the appropriate CodecRegistry instance later
+                        // when the message will be decoded into either a PreparedStatement or a ResultSet
+                        DataType type = DataType.decode(body, protocolVersion, new CodecRegistry());
```

Comment:
should this use DEFAULT_INSTANCE too?  (To prevent creating an instance every time).